### ISSUE TITLE
Custom TileGrid and Projection for Tile Sources

### ISSUE THREAD

**Issue by justb4** [meggsimum/wegue#124]
#82 provided basic support for Maps with a custom projection. A next step is to also support tiled layers with custom OpenLayers' `TileGrids` (other than OSM/Google/Web Mercator global tilegrid).

At least OL `TileLayer Sources` like `XyzSource`, `TileWmsSource` and `VectorTileSource` currently supported in Wegue, support `tileGrid` and `projection` parameters (but no config support yet).

This allows also support for custom TMS (`{-y}`) and WMTS-RESTful Layers.
Regular KVP WMTS is more complex and should be adressed in separate issue "Support KVP WMTS", as a new Layer Type.

Basically the following additions are required:

1) config: global `TileGrid`(one or more?) definitions 
2) config: support `tileGrid` (ref) and `projection` (ref) for `XyzSource`, `TileWmsSource` and `VectorTileSource` Layers
3) Layer Factory: support these new Layer definitions
4) extend the current "projected" example with tiled Layers from Dutch Kadaster/PDOK

**Comment by justb4** [comment]
Working on a PR for this.

**Comment by chrismayer** [comment]
Done with #125
